### PR TITLE
Allow users to create account and sign in with limitied orcid public visibility

### PR DESCRIPTION
**Pull request by davidsmejia**
## Issue Number

#690 

## Purpose/Implementation Notes

* Combines the logic for signing and create account to use the same hook
* handles missing details depending on what comes back from orcid
* makes first name and last name required to create an account on the API
* refactors some of the components in favor of composition

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Functional tests

Tested Sign In and Create Account flows with:

All Open Settings
Private Email
Private Name

## Checklist

- [X] Lint and unit tests pass locally with my changes
- [X] I have added necessary documentation (if appropriate)

## Screenshots

n/a

### CODE REVIEW

File: api/resources_portal/views/user.py [hunk 1]
```diff
@@ -139,102 +167,81 @@ def get_permissions(self):
...
-        last_name = ""
-        if summary["person"]["name"].get("family-name", None):
-            last_name = summary["person"]["name"]["family-name"]["value"]
+        if missing_data:
+            return JsonResponse(
+                {
+                    "error": "There were details not provided on the provided ORCID record. Please provide missing details in the POST request.",
+                    "required": missing_data,
+                },
+                status=401,
```
Comment (kurtwheeler) [inline api/resources_portal/views/user.py:222]: I think a 400 Bad Request makes more sense here than a 401 Unauthorized.

File: client/src/components/LoginRequired.js [hunk 2]
```diff
@@ -29,12 +31,22 @@ export default ({
...
+            ) : (
+              <CreateOrLogin />
+            )}
           </Box>
         </Main>
       </Box>
     )
   }
 
+  // as you were
```
Comment (kurtwheeler) [inline client/src/components/LoginRequired.js:50]: What's this mean?

**Review by kurtwheeler — APPROVED** [review]
This looks good to me!! Nice work with the refactor of create user, that looks wayyyy better now!

File: api/resources_portal/views/user.py (hunk 1 above)
Comment (davidsmejia) [reply api/resources_portal/views/user.py]: updated to use 409 when user exists and 422 when orcid response is missing required info

File: client/src/components/LoginRequired.js (hunk 2 above)
Comment (davidsmejia) [reply client/src/components/LoginRequired.js]: updated comment for clarity

**Review by kurtwheeler — APPROVED** [review]
The additional changes LGTM! Nice work!

**Pull request merged**